beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/hive/HiveMetadataCleaner.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.cleanup.hive;
+
+import com.expediagroup.beekeeper.cleanup.metadata.MetadataCleaner;
+import com.expediagroup.beekeeper.cleanup.monitoring.DeletedMetadataReporter;
+import com.expediagroup.beekeeper.core.config.MetadataType;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.monitoring.TimedTaggable;
+
+public class HiveMetadataCleaner implements MetadataCleaner {
+
+  private HiveClient client;
+  private DeletedMetadataReporter deletedMetadataReporter;
+
+  public HiveMetadataCleaner(HiveClient client, DeletedMetadataReporter deletedMetadataReporter) {
+    this.client = client;
+    this.deletedMetadataReporter = deletedMetadataReporter;
+  }
+
+  @Override
+  @TimedTaggable("hive-table-deleted")
+  public void dropTable(HousekeepingMetadata housekeepingMetadata) {
+    client
+        .dropTableIfExists(housekeepingMetadata.getDatabaseName(), housekeepingMetadata.getTableName());
+    deletedMetadataReporter.reportTaggable(housekeepingMetadata, MetadataType.HIVE_TABLE);
```

Comment:
Do you want to have the above return a boolean and only do the increment if it returned true like the code below? Would be more consistent...

Reply:
Do we want a try/catch around this to also catch errors? Or are we OK with them just being thrown and then I assume the only way to know about them is to look in the log? Or we do track them elsewhere?

Reply:
Thats what it was initially, both the `dropTable` and `dropPartition` were the same and would return true if the drop was successful, which would determine whether it should be reported it. This also got returned to the handler class that calls these methods, as the paths are only deleted if the drop is successful.

I spoke to Patrick and he requested I change it to not return true and actually only call this method from the handler if the table exists. But this doesnt entirely work for the partition b/c you can't check it exists before trying to drop it, so thats why its still returning boolean. 

Theres a try/catch in the `dropTableIfExists` method which will catch+log/ throw depending on the error. If its `NoSuchObj` thats fine so its just logged, anything else will be thrown.

Reply:
OK. Are we OK with just logging the errors and not incrementing a counter that we could potentially put a dashboard and alerts on etc.

Reply:
We're expecting that the only time we'll get those errors is when a table is renamed which creates a new entry in the HousekeepingMetadataRepository, or if a user performs the drop themselves, so it shouldnt be something we need to alert on.

Reply:
OK, let's hope you're correct in your expectations ;)

Reply:
Having alerts on thrown exceptions is a good idea, we do that for the other Beekeeper modules not sure how this is handled now but if it isn't maybe we create a ticket to tackle this as a next step??

Reply:
I was talking about the `NoSuchObj` exceptions, these arent thrown, theyre only logged so I'm not sure its worth counting them. But might be worth counting/alerting on the exceptions that do get thrown. 

I'm not sure how its handled but I see theres a graph for it on the hcom dashboard. It isnt in the EGDL dashboard though. We can add a ticket for this to be added in.

Reply:
I think noSuchObj doesn't need to be counted. General exceptions should similary to what we do with path deletion.